coordinator/src/main/scala/filodb.coordinator/ShardAssignmentStrategy.scala

```diff
@@ -52,15 +56,23 @@ class K8sStatefulSetShardAssignmentStrategy(val maxAssignmentAttempts: Int = 8)
   private val pat = "-\\d+$".r
 
   private[coordinator] def getOrdinalFromActorRef(coord: ActorRef): Option[(String, Int)] =
-    // if hostname is None from coordinator actor path, then its a local actor
-    // If the host name does not contain an ordinal at the end (e.g filodb-host-0, filodb-host-10), it will match None
-    coord.path.address.host
-      .map(host => InetAddress.getByName(host).getHostName)
-      .orElse(Some(InetAddress.getLocalHost.getHostName))
-      .map(name => if (name.contains(".")) name.substring(0, name.indexOf('.')) else name)
-      .flatMap(hostName =>
-        Some((hostName, pat.findFirstIn(hostName).map(ordinal => -Integer.parseInt(ordinal)).getOrElse(-1))))
-
+    Try {
+      implicit val timeout = Timeout(5.seconds)
+      val coordinatorHostName = coord.path.address.host
+        .map(_ => Await.result((coord ? MiscCommands.GetHostName).mapTo[String],
+          FilodbSettings.globalOrDefault.DefaultTaskTimeout)).getOrElse(InetAddress.getLocalHost.getHostName)
```

Comment:
The Await will block the actor dispatcher thread. This is not good, and accumulates tech debt. 
Prefer to move to Clustering V2, but I understand this interim solution may be needed.

Reply:
Thanks @vishramachandran, agreed this isn't an ideal solution. I have added a comment and called it out.